### PR TITLE
add  extension filtered account subscriptions

### PR DESCRIPTION
**Pull request by hamidra**
Re: issue #1062.
Now that  different extensions are adding support for substrate accounts, having some new method like this might help the UIs to let them only subscribe to the specific extensions that the user connects to. Currently, when they subscribe to account changes, they get triggered for the changes from all extensions which might trigger unnecessary re-renders if not handled correctly.

### CODE REVIEW

**Comment by rossbulat** [comment]
This looks great. 
It'll be useful for the [staking dashboard](https://github.com/rossbulat/polkadot-staking-dashboard) that assumes one active wallet at a time. As more extensions roll out this will become more important as to stop the multi pop-up.

**Review by rossbulat — APPROVED** [review]

**Review by rossbulat — APPROVED** [review]

**Comment by jacogr** [comment]
It is in the queue for review, was hoping to get it in for the last point release (or at least get the review done before, if testing was delayed), but alas it was not to be.

Just a general comment from a look-through - keep away from disabling the linter, unless there is a very good reason. For instance for the "floating" promises, log an error if it does happen or have explicitly `.catch((): void => { /* comment why this can NEVER happen ... */})`, no need to re-throw, but these are linter checks in-place ensures we have at least tracking (if something weird does happen) and easy maintenance, i.e. it is explicit in what it does.

**Comment by hamidra** [comment]
@rossbulat, What this new method provides is to  subscribe to account updates for a specific extension,  but the multiple extension pop-ups triggers still remains even with this changes, since that trigger happens during web3Enable(...) call If I'm correct. 
@jacogr Is there any way to only trigger the App Authorization popup for a specific extension?!

File: packages/extension-dapp/src/bundle.ts [hunk 1]
```diff
@@ -177,6 +177,43 @@ export async function web3AccountsSubscribe (cb: (accounts: InjectedAccountWithM
   };
 }
 
+export async function web3ExtensionAccountsSubscribe (extensions: string[], cb: (accounts: InjectedAccountWithMeta[]) => void | Promise<void>, { ss58Format }: Web3AccountsOptions = {}): Promise<Unsubcall> {
```
Comment (jacogr) [inline packages/extension-dapp/src/bundle.ts:180]: A question on this - why don't we just add an optional `extensions?: string[]` param to `web3AccountsSubscribe`? The two are equivalent, just with an additional filter. So would suggest something like -

```js
web3AccountsSubscribe (cb: (accounts: InjectedAccountWithMeta[]) => void | Promise<void>, { extensions, ss58Format }: Web3AccountsOptions = {}) {
...
const unsubs = (await web3EnablePromise)
  .filter(({ name }) =>
    !extensions || extensions.includes(name)
  )
  .map(
    ({ accounts: { subscribe }, name: source }): Unsubcall =>
```

Does that work for you?

Comment (hamidra) [reply packages/extension-dapp/src/bundle.ts:180]: This works very well. I thought we might want to keep extensions out of account options, but this is more DRY and makes sense to have just an extra option there.

Comment (jacogr) [reply packages/extension-dapp/src/bundle.ts:180]: Perfect, all happy with that approach for the same DRY reason. We can add the same to web3Accounts then as well (makes it consistent, it anyway uses the same options interface).

**Review by jacogr — COMMENTED** [review]
Thank you!

LGTM, just waiting for CI to be happy and will merge.

File: packages/extension-dapp/src/bundle.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ export function web3Enable (originName: string, compatInits: (() => Promise<bool
 }
 
 // retrieve all the accounts across all providers
-export async function web3Accounts ({ accountType, ss58Format }: Web3AccountsOptions = {}): Promise<InjectedAccountWithMeta[]> {
+export async function web3Accounts ({ accountType, ss58Format, extensions }: Web3AccountsOptions = {}): Promise<InjectedAccountWithMeta[]> {
```
Comment (jacogr) [inline packages/extension-dapp/src/bundle.ts:112]:
```suggestion
export async function web3Accounts ({ accountType, extensions, ss58Format }: Web3AccountsOptions = {}): Promise<InjectedAccountWithMeta[]> {
```

File: packages/extension-dapp/src/bundle.ts [hunk 3]
```diff
@@ -141,7 +143,7 @@ export async function web3Accounts ({ accountType, ss58Format }: Web3AccountsOpt
   return accounts;
 }
 
-export async function web3AccountsSubscribe (cb: (accounts: InjectedAccountWithMeta[]) => void | Promise<void>, { ss58Format }: Web3AccountsOptions = {}): Promise<Unsubcall> {
+export async function web3AccountsSubscribe (cb: (accounts: InjectedAccountWithMeta[]) => void | Promise<void>, { ss58Format, extensions }: Web3AccountsOptions = {}): Promise<Unsubcall> {
```
Comment (jacogr) [inline packages/extension-dapp/src/bundle.ts:146]:
```suggestion
export async function web3AccountsSubscribe (cb: (accounts: InjectedAccountWithMeta[]) => void | Promise<void>, { extensions, ss58Format }: Web3AccountsOptions = {}): Promise<Unsubcall> {
```

**Review by jacogr — COMMENTED** [review]
Small comments that should pacify CI.

**Review by jacogr — APPROVED** [review]

**Pull request merged**